packages/compile-common/src/index.ts

```diff
@@ -1,2 +1,53 @@
 import { Profiler } from "./profiler";
-export { Profiler };
+import * as shims from "./shims";
+export { Profiler, shims };
+
+import { ContractObject } from "@truffle/contract-schema/spec";
+
+export type Compilation = {
+  sourceIndexes: string[];
+  contracts: CompiledContract[];
+  compiler: {
+    name: string | undefined;
```

Comment:
Should we make `name` and `version` required? I added `undefined` to these because there is a case where compilation doesn't happen and `name` and `version` are `undefined`. Maybe some other solution should be implemented in those cases. I'll note where I'm talking about below.